### DISCUSSION TITLE
Disappearing local data in iPad, how to prevent?

### DISCUSSION THREAD

**Discussion by OMA2k** [airsdk/Adobe-Runtime-Support#1996] [Q&A]
Hello. I'm having an issue with some client who is having an issue with their iPads. My application has an offline mode which downloads a large amount of data into the iPad (one of the files is 22 MB, and the rest are thousands of tiny JPG files). For some reason, this works only for some weeks, then the data automatically disappears!

My app is already using the `file.preventBackup = true;` attribute with ALL files, so why are they still being deleted?

Thanks for reading.

**Comment by marchbold** [comment]
Where are you storing your assets? If you store them in a "cache" location the os may clean them up if the system is running low on disk space.

**Reply by OMA2k** [reply]
I'm storing them in: File.userDirectory.nativePath + '/Library/PrivateDocuments'

This is the code I'm using to reference the storage directory:

```
var storageDir:File = (
	Capabilities.version.substr(0, 3) == 'IOS' ?
		new File(File.userDirectory.nativePath + '/Library/PrivateDocuments') :
		File.applicationStorageDirectory
);

```

I used that iOS discrimination check years ago to comply with Apple's guidelines of where files should be stored, and this code has worked all these years. Should I now use File.applicationStorageDirectory for iOS as well? Does it work well with Apple's guidelines? Would it make any difference?

**Reply by marchbold** [reply]
Hmmm I would have thought the `File.documentsDirectory` was the correct place for this on iOS? Or the `applicationStorageDirectory`. 

I actually thought the user directory was inaccessible on ios but it may have been mapped to some other location.

**Reply by OMA2k** [reply]
I haven't tried either of those. I might try with "File.documentsDirectory" since the "File.applicationStorageDirectory" isn't supposed to store large amounts of data.
